### DISCUSSION TITLE
Wix 3x extension for VS 2022 for ARM

### DISCUSSION THREAD

**Discussion by videoguy** [wixtoolset/issues#8844] [WiX Development]
I understand the recommendation is to move to WiX 4.x or later, but I have a large project built with WiX 3.x. We’re currently working in an environment where developers are using Windows 11 on ARM systems. While the WiX x64 installer for 3x works fine on these systems, we’ve encountered a challenge: there’s no official WiX Visual Studio extension for VS 2022 for ARM.

Here’s the situation:

1. I’ve tried a workaround by copying the necessary WiX folders from an x64 system’s VS directories to the ARM system, preserving the folder structure. This allows me to build our product installers from the CLI without issues. 

Attached is the zip file I created for this.  [wix3xextensionforarm.zip](https://github.com/user-attachments/files/17895022/wix3xextensionforarm.zip)

2. However, Visual Studio 2022 still reports .wixproj files as “incompatible” and doesn’t support building the project from the IDE.

My questions are:
	•	Is there a way to make .wixproj files compatible with Visual Studio 2022 on ARM systems so we can use the IDE for builds?
	•	If not, would it be feasible to build a custom WiX extension for Visual Studio 2022 that works on ARM? Or is there another workaround that might be easier, even if it’s not ideal?

For now, this is intended as a temporary solution until we can prioritize porting to WiX 4.x, which may not happen for another 9+ months. Any advice or suggestions would be greatly appreciated!

Thanks in advance!

**Comment by chrpai** [comment]
Not really.  I just had a customer from Australia go through this.  I quickly updated him to v4 to move on.  I then took the money I billed him and bought myself an ARM64 laptop so I could be more in tune with this arch platform.  It works really good and I like Visual Studio except that there are many VS workloads not available on it. Votive isn't alone.

There's also many other improvements in WiX v4 related to ARM64 support.  The ability to rebuild x86, x64 or ARM64 from the exact same source just a different platform configuration is brilliant.  It motivated me to update my IsWiX tooling to support v4.

If "time" is your only constraint I could have it done before the thanksgiving holidays are over.

**Comment by videoguy** [comment]
Thanks Chris!
What do you suggest I do first?
Just install Wix 4x and install VS 2022 extension for it and build it?

**Comment by chrpai** [comment]
Yes. Votive is replaced with HeatWave. It has a right click convert that usually works pretty well.

If your looking for more of a cleanup/refactoring I'd start here. https://github.com/iswix-llc/iswix-tutorials  I'd be happy to give you a free 1 hour session to demonstrate and answer any questions that you have.

**Comment by chrpai** [comment]
I think I've reconsidered my opinion on this.  I have a bunch of customers. Some are on v3.11 some are on v3.14 and some are on 4.0+.  I recently purchased an ARM64 device because I wanted to make sure all of my tooling (IsWiX) was working perfectly and optimized for ARM64.  It seemed reasonable at first to assume everything would be v4+ but then two scenarios started to happen. 1) I'd get a support request for a customer who was using v3 and I didin't want to have to switch device.  2) I'd be working on IsWiX development and I didn't want to have to switch to a different device to test those legacy designers.

So I took Votive2022.vsix and my knowledge of VS Extensibility and in just a few moments had it hacked up and installed in VS2022 on ARM64.  Everything seems to work just fine.  There might be some heat/harvesting functionality in the .wixproj that doesn't work but I don't use any of that.  I'm not a fan of harvesting unless absolutely needed and the new Files element is pretty much what I'd always choose for that now.

So I'd encourage @robmen  to do a partial release of votive adding ARM64 support to the extension manifest.

**Comment by videoguy** [comment]
Thanks Chris!
I appreciate for giving 3.x extension a try. I love to be your guinea pig to test this on a ARM64 laptop.

**Reply by chrpai** [reply]
Email me. I'm easy to find.

**Comment by videoguy** [comment]
Thanks a lot to Chris for helping me out. https://github.com/fobrs/VisualStudioExtension/releases/tag/v1.0.0.21_arm64 has wix3x extension for vs 2022 on ARM. It worked like charm. I only installed Votive2022.vsix from the zip file on that link. I didn't install Schemas.vsix.
My .wixproj files load and build fine from VS 2022 on ARM systems now.
People who land here from a goolge search can use the link above.

**Reply by chrpai** [reply]
I'd install schemas because that's going to give you intellisense.

**Reply by videoguy** [reply]
Good to know. Thanks
